.github/workflows/ts-packages.yml

```diff
@@ -138,6 +138,66 @@ jobs:
           verbose: true
           flags: sdk
 
+  depcheck:
+    name: Check for unused dependencies
+    runs-on: ubuntu-latest
+
+    steps:
+      - uses: actions/checkout@v2
+
+      - name: Fetch history
+        run: git fetch
+
+      - name: Setup node
+        uses: actions/setup-node@v1
+        with:
+          node-version: '16.x'
+
+      - name: Get yarn cache directory path
+        id: yarn-cache-dir-path
+        run: echo "::set-output name=dir::$(yarn cache dir)"
+
+      - uses: actions/cache@v2
+        id: yarn-cache
+        with:
+          path: ${{ steps.yarn-cache-dir-path.outputs.dir }}
+          key: ${{ runner.os }}-yarn-${{ hashFiles('**/yarn.lock') }}
+          restore-keys: |
+            ${{ runner.os }}-yarn-
+
+      - name: Install Dependencies
+        # only install dependencies if there was a change in the deps
+        # if: steps.yarn-cache.outputs.cache-hit != 'true'
+        run: yarn install
+
+      - name: Check packages/batch-submitter
```

Comment:
I think it's reasonable to assume that all of the packages within `packages` will be npm/yarn based (we're putting Go packages in the `go` directory). In that case, is it possible to loop over each directory within a GitHub action instead of listing every directory explicitly? I haven't worked with GitHub actions enough to know offhand. That way this script would work automatically for new packages and we wouldn't have to remember to add the packages to this list of directories.

Reply:
Hmm, yeah I'm not finding a clean way to do this. There's the option to use a [matrix](https://docs.github.com/en/actions/learn-github-actions/workflow-syntax-for-github-actions#jobsjob_idstrategymatrix), but I don't think it's possible to feed in a dynamic value there.

Reply:
I'll leave myself a reminder to add this to a doc that explains what needs to be modified when adding a new package.